client/components/app.js

```diff
@@ -26,22 +42,20 @@ export class App extends React.Component {
     })
       .then(res => res.json())
       .then(data => {
-        console.log(data)
         this.setState({ list: this.state.list.filter(channel => {
           return channel._id !== twitchId.twitch_id
         }) })
       })
   }
-
+  switchView(param) {
+    this.setState({view: param})
+  }
   render() {
+    const view = determineView(this)
     return (
       <div>
-        <div className="search-bar">
-          <Search getChannel={this.getChannel} />
-        </div>
-        <div>
-          <SearchList list={this.state.list} addChannel={this.addChannel} />
-        </div>
+        <Search getChannel={this.getChannel} switchView={this.switchView}/>
+        {view}
```

Comment:
it's a little out of "React convention" to write it this way (with the `determineView`). Instead, do it inline here:

```jsx
{
  this.state.view === 'Search' 
    ? <SearchList list={component.state.list} addChannel={component.addChannel} />
    : <Favorites favorites />
}
```